app/src/main/java/com/seafile/seadroid2/cameraupload/CameraSyncAdapter.java

```diff
@@ -501,69 +534,120 @@ private String varArgs(int count) {
      * @param cursor
      * @throws SeafException
      */
-    private void iterateCursor(SyncResult syncResult, DataManager dataManager, Cursor cursor, String media) throws SeafException, InterruptedException {
+    private Cursor iterateCursor(SyncResult syncResult, DataManager dataManager, Cursor cursor, String media) throws SeafException, InterruptedException {
 
         tasksInProgress.clear();
         File file;
+        String file_name;
+        String file_path;
+        int file_modified;
+        int file_size;
+        int time_out = 10000;
         // upload them one by one
+        int file_num = cursor.getCount();
+        int file_iter = cursor.getPosition();
         while (!isCancelled() && cursor.moveToNext()) {
+            file_iter += 1;
```

Comment:
Amy reason to keep this comment line?

Reply:
filter_iter is to help us to check whether the camera uploading service has uploaded all images.